src/libraries/Common/src/System/Number.Parsing.Common.cs

```diff
@@ -336,7 +336,7 @@ internal enum ParsingStatus
         private static unsafe TChar* MatchChars<TChar>(TChar* p, TChar* pEnd, ReadOnlySpan<TChar> value)
             where TChar : unmanaged, IUtfChar<TChar>
         {
-            Debug.Assert((p != null) && (pEnd != null) && (p <= pEnd) && (value != null));
+            Debug.Assert((p != null) && (pEnd != null) && (p <= pEnd) && (value != default));
```

Comment:
CC @tannergooding for number changes

Reply:
Changing this from null to default isn't meaningful.

Looking at the PR this came from, it was originally checking that a `string value` wasn't null. Given that, I think the right answer is to just delete this expression from the assert.